### ISSUE TITLE
A better icon.

### ISSUE THREAD

**Issue by xpol** [google/xi-editor#51]

**Comment by raphlinus** [comment]
I was hoping this was a pull request, not just an issue :)

What do you think of this? I just dashed it out in Inkscape, could certainly polish it more.

![xi-logo](https://cloud.githubusercontent.com/assets/242367/14947778/3beeba54-0fed-11e6-82e7-e273a1fb903c.png)

**Comment by xpol** [comment]
Sorry, I'm just a programmer, has no gift in art. 😭

**Comment by rkusa** [comment]
I am also just a programmer and my graphical skills leave much to be desired. However, I'd still like to contribute at least some general ideas. So please consider the following not as final proposals, but just as ideas.

![xi1](https://cloud.githubusercontent.com/assets/409021/14951206/86a4d7ee-1057-11e6-8120-8f3dcad54492.png)
![xi2](https://cloud.githubusercontent.com/assets/409021/14951207/88358ba8-1057-11e6-89ec-7c584ab3d927.png)

Credits: Seen the idea of using highlighted code represented as blocks on a replacement Sublime Text icon some time ago. The colors are from the base16 ocean color scheme. Instead of drawing the blocks by hand, I've used the BLOKK Neue font. "Xi" is written in Source Code Pro. The code used is:

```rust
impl Default for Editor {
    fn default() -> Editor {
        Editor {
            text: Rope::from(""),
            view: View::new(),
            dirty: false,
            delta: Delta::new(),
            new_cursor: None,
            scroll_to: Some(0),
            col: 0
        }
    }
}
```

@raphlinus Does the name *Xi* has a special meaning? If so, the icon could be based around this meaning.

**Comment by disordinary** [comment]
According to Wikipedia Xi is the Greek letter Ξ or ξ it's also the Romain numerals for 11.

The thing to think about is what will set Xi apart from the other editors that exist; as far as I can gather it's mainly being platform native, the client/server architecture, and the performance goals. And from an end users perspective its the native ui and performance that will set it apart from something like atom.

With the above in mind and considering raphlinus's background I'd suggest simplicity with a beautiful font.

**Comment by leavengood** [comment]
I've had similar thoughts about a new icon. I looked up Xi as @disordinary has done and thought the uppercase Ξ would make a nice element of the icon. I also like the general direction @rkusa is going. My customized Sublime Text icon (https://dribbble.com/shots/1827862-Yosemite-Sublime-Text-Icon) uses these colors. Personally I don't like using the name Xi in the icon though.

Maybe the icon could be a circle with a large Ξ in the middle, with a subtle interaction showing between the upper and lower bars through the middle one, representing the front and back ends communicating though RPC. I can see what I can throw together in Sketch.

It goes without saying probably, but a discussion like this is the perfect [bikeshed](https://shed.bike) type of discussion.

We also may want to discuss the name and pronunciation of Xi ("zigh" from what I've seen) in the README (but I digress.)

**Comment by chriskrycho** [comment]
I just came here to link the same Sublime Text icon family @leavengood linked, so mission accomplished there. A stylized use of Ξ would be super interesting; I'd be happy to look around for typefaces with interesting variations on the Greek letters and report back.

**Comment by gaganjyot** [comment]
Something like Use of Ξ enclosing in a gear ( just like in rust logo, https://blogs.s-osg.org/wp-content/uploads/2015/04/rust-logo-blk.png ) would also be nice.

P.S Again just a programmer. No Ui skills.

**Comment by rkusa** [comment]
The icons @leavengood linked to are similar to the ones I meant (maybe even better). Actually, I would also prefer to not have the name "Xi" as part of the icon. Something similar to the linked icons but a **Ξ** instead of the **S** sounds good

**Comment by raphlinus** [comment]
+1 to one of the @leavengood designs with a Ξ.

**Comment by leavengood** [comment]
I noticed that there was a Sketch file for the Sublime icon I linked to, so I modified it a bit and this is the result:

![xi_icon2_2](https://cloud.githubusercontent.com/assets/637/14963146/5907b18a-1070-11e6-9a5e-63ef6e4ad4e8.png)

It's a bit blurry because I exported the 1024 px version to 256 px.

I made the Ξ bigger than the S in the original icon, and I brought down the opacity of the block source code in the back since the Ξ was getting lost.

I don't know if this is the direction we want to go, but I figured I would put it out there. Probably we would need permission from the original artist, otherwise I should start fresh.

**Comment by leavengood** [comment]
Oh I also just realized I may have suggested those Sublime icons were my creation, they are not, I just make use of them as my Sublime icon. Sorry for the confusion. Again we would need to contact the artist on Dribbble to get proper permission to make use of a sligthly modified version.

**Comment by chriskrycho** [comment]
Worth note that using a serif or slab serif with a Ξ could make a very nice effect. Here's Irma Text Slab Pro, for example:

![Irma Text Slab Pro Light Ξ](https://cloud.githubusercontent.com/assets/2403023/14963327/1f3aeb88-1071-11e6-8272-ce510667409a.png)

And Brioni Text Pro:

![Brioni Text Pro Medium](https://cloud.githubusercontent.com/assets/2403023/14963371/510c6948-1071-11e6-92b2-00d918b41cc6.png)

Both of these give you both the *Ξ* character and a fairly code-like glyph. (Note that I'm not so much suggesting either of these, though both are quite nice, as I am suggesting a direction to consider.)

**Comment by leavengood** [comment]
Those are pretty cool @chriskrycho, definitely worth considering, especially the first one. I just checked and the icon I posted is using Helvetica Neue, changing to Source Code Pro changes the size of the lines a bit (the middle one is smaller.) Here is a crazy version using Copperplate:

![xi_icon_copperplate](https://cloud.githubusercontent.com/assets/637/14963603/77690a3c-1072-11e6-8610-afb6987acc84.png)

**Comment by rkusa** [comment]
I like both (especially the first one), because they are more interesting than just three lines. However, after looking at them for a while, they kinda look like hourglasses. If I am not the only one seeing hourglasses, this may be contrary to the performance goal of Xi.

**Comment by chriskrycho** [comment]
The other real downside to that approach is the (limited but real) similarity to the Brackets icon:

![Brackets icon](https://cloud.githubusercontent.com/assets/2403023/14964000/633d2c4e-1074-11e6-87cb-165898951243.jpg)

**Comment by chriskrycho** [comment]
More-or-less stylized lowercase ξ can also be interesting. Here's Parmigiano Headline Pro Bold Italic:

![Parmigiano Headline Pro Bold Italic](https://cloud.githubusercontent.com/assets/2403023/14964090/c5107962-1074-11e6-8396-e513a791a598.png)

(The overlaid gray text is a clue where I'm snagging some of these; I'm using [FontStand](https://fontstand.com), which is awesome.)

**Comment by Thourum** [comment]
I gave it a try. Tell me what you think.
<img width="512" alt="xi" src="https://cloud.githubusercontent.com/assets/12977004/14964565/fba1e84c-10a8-11e6-86aa-84fbd28dae77.png">

**Comment by leavengood** [comment]
Yeah I kind of "see" the hourglass in those serif fonts as well.

Here is a slightly more interesting version using "Hiragino Kaku Gothic Std", which has a Japanese flavor:

![xi_icon_japanese_font](https://cloud.githubusercontent.com/assets/637/14964502/5e0d085a-1076-11e6-8a39-52710f4ea477.png)

And with that I need to focus on other things for a while :sweat_smile:

**Comment by ywecur** [comment]
My immediate reaction when I saw Ξ was to think of the hamburger menu. Probably not the best impression you'd want to give. I'd suggest using ξ in the logo instead, if not simply Xi like @rkusa did.

**Comment by leavengood** [comment]
This is an interesting version which came up when I was switching through fonts in Sketch (this is still the uppercase Ξ):

![xi_icon_zz](https://cloud.githubusercontent.com/assets/637/14970717/7104ebc6-1099-11e6-92bd-26c91d94e2fc.png)

The font is "Lantinghei SC".

As for your comment about the hamburger menu @ywecur, I can see your point but it isn't necessarily a bad connotation. But nonetheless here is a version with the lowercase ξ (in Source Code Pro):

![xi_icon_lowercase](https://cloud.githubusercontent.com/assets/637/14970826/43d0abc6-109a-11e6-9812-f9f24cf4b6ae.png)

Also I like aspects of @Thourum's icon, especially the brighter source code in the background. I could try doing that and making the shadow bigger in my version.

**Comment by ywecur** [comment]
@leavengood You're right that it's not inherently bad, but it's unfitting for a text editor. A hamburger menu raises connotations of mobile UIs, which isn't the feel you'd want from a powerful and **fast** text editor. Just my ¢2.

**Comment by ywecur** [comment]
I'm not good at image editing but a logo style similar to that of PyCharm would be fitting, I believe.

![](http://blog.jetbrains.com/pycharm/files/2015/12/PyCharm_400x400_Twitter_logo_white.png)

**Comment by Thourum** [comment]
I tried several designs with " ξ " but in many of my designs it looks very similar to the  Emacs logo

![1024px-emacsicon svg](https://cloud.githubusercontent.com/assets/12977004/14976829/62bd82b4-110f-11e6-98d1-7e75c8f92f80.png)

**Comment by PixelPirate** [comment]
What purpose will the Icon posted here have? Is it just a neat icon for the xi-core github project? 

Because one of the goals of Xi is to have beautiful, native felling front-end apps for each platform. And each platform has it's own guidelines for app icons.

I would expect xi-cocoa to have a icon conforming to the Mac OS guidelines, but for xi-windows to have a icon conforming to the rest of the icons in Windows 10.

Instead of discussing an actual icon, I suppose it would be better to discuss a common glyph or ornament that each platform specific icon should implement, in order for each platform specific icon to look like being of the "Xi-Family".

**Comment by disordinary** [comment]
So it's more branding and a logo. I think that whatever is decided here would be temporary if the project takes off. The main thing is to think about what is wanted to be represented in the logo from a branding point of view and what it represents, rather than saying this is how something else does it lets copy that.

**Comment by Thourum** [comment]
The " Xi-Family " is a good idea, however having 2 different icons for the same editor might be confusing.

Discussing the common design guidelines can be discussed later.

**Comment by PixelPirate** [comment]
I don't think different icons are confusing...
![os-x-el-capitan-ibooks-for-mac-icon-full-size](https://cloud.githubusercontent.com/assets/776450/14979888/790016ca-1125-11e6-9365-f05182fbafce.jpg)

But anyway, this project is still very young, and there no harm in having a single icon for everything in the beginning, I was just thinking out loud for what could be the direction in the future :)

**Comment by mblode** [comment]
![xi 2](https://cloud.githubusercontent.com/assets/7183998/14982099/11ce63ec-1177-11e6-8121-ff11d9df6ce7.png)
![xi](https://cloud.githubusercontent.com/assets/7183998/14982100/11ceebd2-1177-11e6-9036-769bf2a21059.png)

Some basic Yosemite inspired icons. Green or Space grey options. The Xi logo is in Raph Levien's Insonsolata monospaced font. I am very open to feedback.

**Comment by stoivo** [comment]
@mblode, can you try to make one version where the border is a gear? Like this just bigger 
![screen shot 2016-05-03 at 14 18 18](https://cloud.githubusercontent.com/assets/3492040/14982890/0a041f2c-113a-11e6-9562-dacb24425d14.png)

**Comment by leavengood** [comment]
I think it is better to be inspired by the Rust logo gear than to use it directly. As for the general discussion of logos/icons, there definitely needs to be a general consistent theme if not the same icon across operating systems. I think the Chrome icon is a good example of the same icon used across different operating systems.

Lastly I had another general idea for an icon, I just need to take some time to experiment with it: an icon that is split in some way (probably diagonally) with a stylized editor on the top and exposed gears on the bottom. Sort of like seeing inside a watch.

**Comment by PixelPirate** [comment]
Here is the icon of [VimR](https://github.com/qvacua/vimr), just as a source of inspiration maybe.
<img width="256" alt="asd" src="https://cloud.githubusercontent.com/assets/776450/14983701/1c17c682-113f-11e6-8e7c-2ee41bb1ddf9.png">

**Comment by Thourum** [comment]
Here are some of the "Designs" i mentions before.
<img width="512" alt="xi_logos" src="https://cloud.githubusercontent.com/assets/12977004/14985346/6c5ca3da-1147-11e6-9e31-6487ba89490e.png">

- 	**Myriad Pro Regular**
-	**Osaka Regular**
-	**Wawati TC Regular**
-	**Nanum Gothic Regular**

**Comment by aratno** [comment]
What about something like this? I threw it together quickly, and I'd be open to any criticism.

![draft01](https://cloud.githubusercontent.com/assets/8835036/14992528/68a29366-1134-11e6-9d09-66429a2ab5a9.png)

**Comment by baskerville** [comment]
![xi-editor](https://cloud.githubusercontent.com/assets/159419/15013593/d3356966-1200-11e6-84a5-a8d0d68df6b1.png)

**Comment by Canode** [comment]
Hehe------just a programmer.

**Comment by Ephil012** [comment]
I think @mblode 's icon is probably the most consistent with OSX, but there should only be one icon for the editor.

**Comment by chriskrycho** [comment]
Actually, the variety of styles on OS X is _much_ wider than just the sort of glossy bubbles in use in some of the main OS icons, and even there there's a wide variety. Going for an utter _lack_ of distinguishability from other generic icons is probably not the best move; finding something with some personality is a much better move.

For example, Bee's icon is *incredibly* distinctive, and feels right at home on the Mac:

![Bee app](https://cloud.githubusercontent.com/assets/2403023/15050464/fd9606ea-12c1-11e6-8c4e-f8d0214a3dd2.png)

My point isn't to go for something like that, but rather to clarify that "circle with a gradient" isn't more or less native-seeming on OS X. 😄

**Comment by leavengood** [comment]
Indeed, circle with a gradient is *easy* and it provides some constraints, but it is also a bit boring.

FWIW I'm slowly working on another icon idea which I do intend to generally make round, but made of two half circles at a 45 degree angle, one being a stylized editor (reminiscent of my and Thorum's icons above) and the other side being a gear, with something connecting them. The idea being to represent the core Xi idea of a "pretty frontend connecting to a hard-working, fast backend".

I'll see how it goes and post it here when I have a decent prototype. If anyone else wants to take that idea and run with it, please do.

**Comment by leavengood** [comment]
OK, here is a really rough version of my icon concept. The gear is ugly (and more of a badge or star than a gear) and probably the editor side could be better, but this gives the idea:

![editor_gear](https://cloud.githubusercontent.com/assets/637/15051537/23de5672-12c7-11e6-95f3-ce5e17cfcd39.png)

We could do the same concept rectangular with the two halves on top and bottom too.

**Comment by vmdanilov** [comment]
![xi-color](https://cloud.githubusercontent.com/assets/3129436/15092053/fabb7a6e-146e-11e6-9721-a6123a5888c9.png)

![xi-color-small](https://cloud.githubusercontent.com/assets/3129436/15092057/21b0f176-146f-11e6-9e12-95e2ec199f33.png)

**Comment by pathawks** [comment]
If there are colored lines of text (like in the Sublime icon), what if lines of a certain color formed the shape of Ξ?

**Comment by liutikas** [comment]
![xi-1](https://cloud.githubusercontent.com/assets/1418185/15096279/b7544aac-14a6-11e6-81cc-fd19896c3345.png)

Here is my proposal. I think we should have something clean that could pass a squint test and would look good in different colors/backgrounds.

**Comment by ywecur** [comment]
I give my strongest vote to @vmdanilov ! That style of icon looks awesome!

**Comment by raphlinus** [comment]
I like it too, but is it too close to the Visual Studio Code logo?

I love seeing the ideas and the discussion. I think there are several here that would work just fine. One of my favorites is @leavengood's with the Source Code pro ξ, but I like several of the others as well. Everyone please feel free to use the thumbs up and thumbs down, and if a consensus favorite emerges, we'll go with that.

**Comment by bnealey** [comment]
![image](https://cloud.githubusercontent.com/assets/5580290/15327768/84d2df22-1c20-11e6-9784-1c91735a2f54.png)

**Comment by bnealey** [comment]
![image](https://cloud.githubusercontent.com/assets/5580290/15327832/ce99ed44-1c20-11e6-9d9e-7a11946b5af5.png)

**Comment by bnealey** [comment]
![image](https://cloud.githubusercontent.com/assets/5580290/15327996/6abdabc0-1c21-11e6-9ff4-f46412d83508.png)

**Comment by phoenixenero** [comment]
![ss 2016-05-24 at 01 44 15](https://cloud.githubusercontent.com/assets/9031092/15493296/b0d86c0a-21b5-11e6-9a8e-98a55132c011.png)

Turn the `Xi` character on its side and you'll get two square brackets and an I. Just a rough sketch.

**Comment by eyelash** [comment]
![xi 02](https://cloud.githubusercontent.com/assets/2668288/15655530/3f732ecc-269e-11e6-8f99-e223d26cf18d.png)

**Comment by eyelash** [comment]
![xi 03](https://cloud.githubusercontent.com/assets/2668288/15656124/55a5a3cc-26a4-11e6-8e90-0ccdbd4ed701.png)

**Comment by mblode** [comment]
@eyelash Could you create that last logo with the same background colours as this: https://cloud.githubusercontent.com/assets/7183998/14982100/11ceebd2-1177-11e6-9036-769bf2a21059.png

**Comment by eyelash** [comment]
@mblode I will give it another try tomorrow and make it a bit greener :smiley:

**Comment by eyelash** [comment]
Here is a tweaked version also with some colour variations:
![xi 04](https://cloud.githubusercontent.com/assets/2668288/15673973/740c3862-273a-11e6-9329-c33056498654.png)
![xi 04 02](https://cloud.githubusercontent.com/assets/2668288/15673988/8fd88168-273a-11e6-9335-919548201398.png)
![xi 04 03](https://cloud.githubusercontent.com/assets/2668288/15674015/bdc2b1e8-273a-11e6-9c1f-e8fdad82e5d4.png)
![xi 04 04](https://cloud.githubusercontent.com/assets/2668288/15674018/c247df90-273a-11e6-8b2c-e7c82805118b.png)
The colours I used are the following:

original:
HSV(150°, 70%, 80%) to HSV(165°, 70%, 90%)

tweaked greener version:
HSV(175°, 70%, 80%) to HSV(90°, 70%, 90%)

blue:
HSV(240°, 50%, 80%) to HSV(270°, 50%, 90%)

brown:
HSV(30°, 50%, 80%) to HSV(45°, 50%, 90%)

You can find the source [here](https://gist.github.com/eyelash/7ea23bbd9aec3e4743cee82ee8d4e1cc) in case anyone wants to do some more tweaks.

**Comment by dragostis** [comment]
I've added a small [PR](https://github.com/google/xi-editor/pull/87), not knowing about the issue. (Xi is growing incredibly much)

I was thinking about a cursive design that incorporates a cursor:

<img src="https://cloud.githubusercontent.com/assets/4136413/16412168/734f9158-3d33-11e6-9bf0-44387fb8c502.png" width="256px" height="256px"/>

**Comment by peteyy** [comment]
@dragostis this is most certainly the best fitting logo to the current macOS design guidelines and I think it's a great idea to make a different logo for every platform

**Comment by dragostis** [comment]
@peteyy Slightly different designs is most definitely a great idea. I personally think that it would be great to have a simple logo that is easy to use in any context and that is easily recognizable, apart from being esthetically pleasing and creative.

**Comment by thomas-profitt** [comment]
I'd like to add that the logo and application icon don't necessarily have to be one in the same. We can have a single _logo_ for xi itself  -- xi-core and the xi project -- which is immediately recognizable and platform-agnostic, and leave _application icons_, which take into account design conventions of the platform on which they run, to the xi front-end applications.

Otherwise, xi could end up with a logo and application icon that not only looks appropriate on some platforms but not others, but also visually connects xi-core to a specific platform, which goes against the model.

**Comment by yuutayamada** [comment]
Please ignore my comment if it's not related to this icon's discussion. (I just found this issue while I was discovering this repo and I'm an Emacs user)

> According to Wikipedia Xi is the Greek letter Ξ or ξ it's also the Romain numerals for 11.

The character "Ξ" is similar to Japanese and Chinese character "三" (which means three), so for those people the Ξ icon might not recognizable as Xi.

**Comment by ywecur** [comment]
@raphlinus No, I don't think so. Though if you feel strongly about it then maybe change the color?

**Comment by adrientetar** [comment]
FWIW, here's an icon I made for the Android frontend I am working on:

![web_hi_res_512](https://cloud.githubusercontent.com/assets/1250894/22380648/717385f0-e4be-11e6-8e64-54ed932d96c1.png)

**Comment by TUSF** [comment]
If we want to use a Ξ in the design, how about something like so:
![xi-glyph](https://cloud.githubusercontent.com/assets/8594356/25510705/cb125da6-2b87-11e7-9bd9-1089e3616872.png)

It sorta looks like X[i], where `i` is a common iteration variable. I also agree with some of the above comments, that instead of making a whole icon, it might be better to create a single glyph that could be adapted to multiple design layouts.

**Comment by xeoneux** [comment]
I love the hexagon icon by @rkusa

**Comment by balthild** [comment]
@vmdanilov

> ![xi-color-small](https://cloud.githubusercontent.com/assets/3129436/15092057/21b0f176-146f-11e6-9e12-95e2ec199f33.png)

By considering different polygons as different "layers of acrylic" and adding some shadows and corner radius, it may conform Fluent Design System, the new look of icons in Win10.

![图片](https://user-images.githubusercontent.com/2662758/77230995-071f8880-6bd3-11ea-973b-ada8b432cb8c.png)

**Comment by facekapow** [comment]
Hey there, I'm the author of the current icon (#16). I originally stumbled upon Xi when it was in its infancy and contributed the icon just so it would have one. I'm a programmer, not a designer, and I whipped something up in under an hour; I think it's safe to say it's certainly not a good icon. I just recently came across this project again and noticed it still had that icon. Honestly, I'm a little embarrassed it's still there, it's pretty bad :sweat_smile:.

I think that the green or grey icons on https://github.com/xi-editor/xi-editor/issues/51#issuecomment-222673740 would look right at home on macOS (though they would need to be updated for macOS's new icon styles), but I think that a good cross platform icon would be the hexagon version on https://github.com/xi-editor/xi-editor/issues/51#issuecomment-216169829.
